static/app/views/settings/account/apiTokens.tsx

```diff
@@ -6,118 +6,141 @@ import {
+          API_TOKEN_QUERY_KEY,
+          oldTokenList => {
+            return oldTokenList?.filter(tk => tk.id !== token.id);
+          }
+        );
+
+        return {previous};
+      },
+      onSuccess: _data => {
+        addSuccessMessage(t('Removed token'));
+      },
+      onError: (_error, _variables, context) => {
+        addErrorMessage(t('Unable to remove token. Please try again.'));
+
+        if (context?.previous) {
+          setApiQueryData<InternalAppApiToken[]>(
+            queryClient,
+            API_TOKEN_QUERY_KEY,
+            context.previous
+          );
+        }
+      },
+      onSettled: () => {
+        queryClient.invalidateQueries(API_TOKEN_QUERY_KEY);
+      },
+    }
+  );
 
-  getDefaultState() {
-    return {
-      ...super.getDefaultState(),
-      tokenList: [],
-    };
+  if (isLoading) {
+    return <LoadingIndicator />;
   }
 
-  getEndpoints(): ReturnType<DeprecatedAsyncView['getEndpoints']> {
-    return [['tokenList', '/api-tokens/']];
+  if (isError) {
+    return <LoadingError onRetry={refetch} />;
   }
 
-  handleRemoveToken = (token: InternalAppApiToken) => {
-    addLoadingMessage();
-    const oldTokenList = this.state.tokenList;
-
-    this.setState(
-      state => ({
-        tokenList: state.tokenList?.filter(tk => tk.id !== token.id) ?? [],
-      }),
-      async () => {
-        try {
-          await this.api.requestPromise('/api-tokens/', {
-            method: 'DELETE',
-            data: {tokenId: token.id},
-          });
-
-          addSuccessMessage(t('Removed token'));
-        } catch (_err) {
-          addErrorMessage(t('Unable to remove token. Please try again.'));
-          this.setState({
-            tokenList: oldTokenList,
-          });
-        }
-      }
-    );
-  };
-
-  renderBody() {
-    const {tokenList} = this.state;
-
-    const isEmpty = !Array.isArray(tokenList) || tokenList.length === 0;
-
-    const action = (
-      <Button
-        priority="primary"
-        size="sm"
-        to="/settings/account/api/auth-tokens/new-token/"
-        data-test-id="create-token"
-      >
-        {t('Create New Token')}
-      </Button>
-    );
-
-    return (
-      <div>
-        <SettingsPageHeader title={this.getTitle()} action={action} />
-        <TextBlock>
-          {t(
-            "Authentication tokens allow you to perform actions against the Sentry API on behalf of your account. They're the easiest way to get started using the API."
-          )}
-        </TextBlock>
-        <TextBlock>
-          {tct(
-            'For more information on how to use the web API, see our [link:documentation].',
-            {
-              link: <ExternalLink href="https://docs.sentry.io/api/" />,
-            }
+  const isEmpty = !Array.isArray(tokenList) || tokenList.length === 0;
+
+  const action = (
+    <Button
+      priority="primary"
+      size="sm"
+      to="/settings/account/api/auth-tokens/new-token/"
+      data-test-id="create-token"
```

Comment:
nit: do we need a test-id here? are we using it?

Reply:
nice catch! Was already unused before my refactoring. Will remove it! 👍